server/blindai_sgx/src/model_store.rs

```diff
@@ -10,12 +10,11 @@ use std::{
     collections::{hash_map::Entry, HashMap},
     sync::Arc,
 };
-use uuid::Uuid;
 
 use crate::model::{InferenceModel, ModelDatumType, OnnxModel};
 
 struct InnerModelStore {
-    models_by_id: HashMap<Uuid, InferenceModel>,
+    models_by_nmid: HashMap<Option<String>, InferenceModel>,
```

Comment:
why would you want an entry for None in the hashmap

HashMap<String, ...> would be much better